### DISCUSSION TITLE
Problems with installing Milvus on multiple servers

### DISCUSSION THREAD

**Discussion by zyy20191** [milvus-io/milvus#18224] [Q&A and General discussion] [answered]
1.Will the distributed version perform better than the standalone version?

2.According to the documents on the official website, I only found different commands through docker installation, but they are essentially on the same machine, aren't they?If there is only one server, will there be any difference in performance between distributed and standalone?

3.Install distributed edition on multiple servers, how to link them together, is there any tutorial documentation?

**Comment by yhmo** [comment, marked as answer]
Both [standalone](https://milvus.io/docs/v2.0.x/install_standalone-docker.md) and [cluster](https://milvus.io/docs/v2.0.x/install_cluster-docker.md) have "Docker Compose", deploy Milvus on a single host. But "Docker Compose" is not recommended for cluster mode.
If you want to deploy cluster on multiple hosts, start a node on each host, you need to split the docker-compose.yaml into several pieces. You need to do some special configuration to make sure these nodes can access each other by correct IP/Port. And if one node is down, you need to manually restart it.
We mainly use "Docker Compose" to deploy cluster for testing and demo. If you want to deploy milvus cluster for production, you ought to use Kubernetes.

**Reply by zyy20191** [reply]
So when I only have one server in production, the performance of both versions is the same, right?

**Reply by yhmo** [reply]
> So when I only have one server in production, the performance of both versions is the same, right?

Deploy standalone on one server, there is only one process, some of the interactions between node/coord is in-process invocation.
Deploy cluster on one server, there are several processes running on the server, the interactions between node/coord is remote rpc invocation.
So I think the performance of "cluser on one server" is a bit worse than "standalone".

**Reply by yhmo** [reply]
And cluster uses Pulsar as a third-party service to transfer data and requests. Standalone doesn't use Pulsar, it uses in-process RocksDB to do the work. So this is another factor that the performance of "cluser on one server" is a bit worse than "standalone".
